sdks/java/core/src/test/java/org/apache/beam/sdk/io/FileBasedSinkTest.java

```diff
@@ -397,6 +417,59 @@ public void testGenerateOutputFilenamesWithoutExtension() {
     assertEquals(expected, actual);
   }
 
+  @Test
+  public void testDisplayData() {
+    FileBasedSink<?> sink = new SimpleSink("gs://bucket/foo/", "xml", "bar-NNN");
+    assertThat(DisplayData.from(sink), hasDisplayItem(allOf(
+        hasKey("fileNamePattern"),
+        hasValue("gs://bucket/foo/bar-NNN.xml"),
+        hasLinkUrl(GcsPath.fromUri("gs://bucket/foo/").getBrowseUrl())
+    )));
+  }
+
+  @Test
+  public void testDisplayDataLinkUrl() throws IOException {
+    class TestCase {
+      String prefix;
+      String shardTemplate;
+      String suffix;
+      String expectedBrowsePath;
+
+      TestCase(String prefix, String shardTemplate, String suffix, String expectedBrowsePath) {
+        this.prefix = prefix;
+        this.shardTemplate = shardTemplate;
+        this.suffix = suffix;
+        this.expectedBrowsePath = expectedBrowsePath;
+      }
+    }
+
+    Iterable<TestCase> tests = ImmutableList.<TestCase>builder()
+        .add(new TestCase("gs://bucket/foo/", "bar", ".xml", "gs://bucket/foo/bar.xml"))
```

Comment:
This looks like a test that would be better served by being [parameterized](https://github.com/junit-team/junit4/wiki/Parameterized-tests)